src/js/redux_state/modules/registration.js

```diff
@@ -47,91 +49,113 @@ export const actions = makeActions('registration', {
         }))
 
         if (entropy.isReady()) {
-          const randomString = entropy.getRandomString(4)
-          return dispatch(actions.submitEntropy(randomString))
+          return dispatch(actions.setRandomString({
+            randomString: entropy.getRandomString(4)
+          }))
         }
       }
     }
   },
 
-  submitEntropy: {
-    expectedParams: ['randomString'],
-    creator: (randomString) => {
-      return (dispatch, getState) => {
-        const entropyState = getState().getIn([
-          'registration',
-          'passphrase',
-          'sufficientEntropy'
-        ])
-
-        if (!entropyState) {
-          throw new Error('Not enough entropy!')
-        }
-        dispatch(actions.generateSeedPhrase(randomString))
-      }
-    }
-  },
-
-  generateSeedPhrase: {
-    expectedParams: ['randomString'],
-    creator: (randomString) => {
-      return (dispatch, getState) => {
-        const mnemonic = bip39.entropyToMnemonic(randomString)
-        dispatch(actions.setPassphrase({mnemonic}))
-      }
-    }
-  },
-
   generateAndEncryptKeyPairs: {
     expectedParams: [],
     async: true,
-    creator: (params) => {
+    creator: () => {
       return async (dispatch, getState, {services, backend}) => {
-        const seedPhrase = getState().getIn([
+        const randomString = getState().getIn([
           'registration',
           'passphrase',
-          'phrase'
+          'randomString'
         ])
-        if (!seedPhrase) {
-          throw new Error('No seedphrase found.')
-        }
-        const masterKeyPair = deriveMasterKeyPairFromSeedPhrase(seedPhrase)
-        console.log(masterKeyPair, 'here is our masterkeypair')
-        const genericSigningKey = deriveGenericSigningKeyPair(masterKeyPair)
 
         const password = getState().getIn([
           'registration',
           'encryption',
           'pass'
         ])
 
+        if (!randomString) {
+          // TODO consistent error handling
```

Comment:
Yes! And I think we have an issue for that opened

Reply:
Yes, we'll need to tackle this soon.